x/mongo/driver/session/client_session.go

```diff
@@ -262,6 +274,30 @@ func (c *Client) UpdateRecoveryToken(response bson.Raw) {
 	c.RecoveryToken = token.Document()
 }
 
+// UpdateSnapshotTime updates the session's value for the atClusterTime field of ReadConcern.
+func (c *Client) UpdateSnapshotTime(response bsoncore.Document) {
+	if c == nil {
+		return
+	}
+
+	subDoc := response
+	if cur, ok := response.Lookup("cursor").DocumentOK(); ok {
```

Comment:
Is there a case where the `"atClusterTime"` value is not in the `"cursor"` document?

Reply:
For [disinct commands](https://github.com/mongodb/specifications/blob/master/source/sessions/snapshot-sessions.rst#server-command-responses) it's in the top level document and not in the cursor subdocument.